firefox-ios/Client/TabManagement/WindowTabsSyncCoordinator.swift

```diff
@@ -0,0 +1,55 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Shared
+import Common
+import Storage
+
+protocol WindowTabsSyncCoordinatorDelegate: AnyObject {
+    /// Returns a collection of all tab managers for which tabs
+    /// should be collected for syncing with the user's profile.
+    func tabManagers() -> [TabManager]
+}
+
+final class WindowTabsSyncCoordinator {
+    private struct Timing {
+        static let throttleDelay = 0.5
+        static let dbInsertionDelay = 0.1
+    }
+    private let throttler = Throttler(seconds: Timing.throttleDelay)
+    weak var delegate: WindowTabsSyncCoordinatorDelegate?
+    private let profile: Profile
+    private let logger: Logger
+
+    init(profile: Profile = AppContainer.shared.resolve(),
+         logger: Logger = DefaultLogger.shared) {
+        self.profile = profile
+        self.logger = logger
+    }
+
+    func syncTabsToProfile() {
+        throttler.throttle { [weak self] in self?.performSync() }
```

Comment:
I would call performSync directly until FXIOS-9050 is done. This is spammed a bunch so the risk that it won't ever execute isn't worth the optimization.

Reply:
> I would call performSync directly until [FXIOS-9050](https://mozilla-hub.atlassian.net/browse/FXIOS-9050) is done.

I should be able to pick up that task sometime soon; in the meantime I've updated this to call `performSync`; there shouldn't be any functional difference between that and the previous code while in single-window mode.